src/app/subjects/subject-form/subject-form.component.ts

```diff
@@ -0,0 +1,78 @@
+import { Component, OnInit, ViewChild } from '@angular/core';
+import { FormGroup } from '@angular/forms';
+import { Router } from '@angular/router';
+
+import { Subscription, Observable, of } from 'rxjs';
+
+import { SubjectFormService } from '../services/subject-form.service';
+import { FormComponent } from '../../shared/components/index';
+import { IFormConfig } from 'src/app/common/models/Form';
+import { Subject } from '../../common/models/Subject';
+import { SubjectService } from '../services/subject.service';
+import { ConfirmSaveService } from '../../common/services/index';
+import { IConfirmSave } from '../../common/models/Confirm-save';
+
+@Component({
+  selector: 'app-subject-form',
+  templateUrl: './subject-form.component.html',
+  styleUrls: ['./subject-form.component.scss'],
+  providers: [SubjectFormService, ConfirmSaveService],
+})
+export class SubjectFormComponent implements OnInit {
+  private isSaving: boolean;
+  @ViewChild('form')
+  public form: FormComponent;
+  public config: IFormConfig;
+
+  constructor(
+    private formService: SubjectFormService,
+    private subjectService: SubjectService,
+    private confirmSave: ConfirmSaveService<Subject>,
+    private router: Router,
+  ) {
+    this.isSaving = false;
+  }
+
+  public ngOnInit(): void {
+    this.config = this.formService.config;
+
+    // set clear function for form
+    this.config.buttons[1].onClick = () => {
+      const { form } = this.form;
+
+      form.reset();
+      this.subjectService.removeSubjectStorage();
+    };
+  }
+
+  public onSubmit(data: FormGroup): void {
+    this.isSaving = true;
+
+    const subject: Subject = this.formService.getSubjectOfForm(data);
+    subject.name = subject.name.toLowerCase();
+    const subscription: Subscription = this.subjectService.addSubjectServer(subject)
```

Comment:
It is fine not to assign to this variable at all const subscription: Subscription